components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/MobileNumberVerificationHandler.java

```diff
@@ -251,7 +255,7 @@ private void preSetUserClaimOnMobileNumberUpdate(Map<String, String> claims, Use
 
         String mobileNumber = claims.get(IdentityRecoveryConstants.MOBILE_NUMBER_CLAIM);
 
-        if (StringUtils.isNotBlank(mobileNumber)) {
+        if (StringUtils.isNotBlank(mobileNumber) && isMobileVerificationOnUpdateEnabled(user.getTenantDomain())) {
```

Comment:
Shouldn't this be `isVerificationPendingMobileClaimConfigAvailable(user.getTenantDomain())` ?